utils/aws/route53/__init__.py

```diff
@@ -0,0 +1,318 @@
+        return self._name
+
+    @property
+    def records(self):
+        """Return DNS records"""
+        return {f"{r.name}": r for r in self._records}
+
+    def add_record(self, record):
+        """
+        Add a record to the zone
+
+        :param record: the record to add
+        :type record: Record
+        :raises DuplicateException: if a record by that name already exists
+        """
+        if self.records.get(record.name):
+            raise DuplicateException(record, self)
+        self._records.append(record)
+
+    def __eq__(self, other):
+        if not isinstance(other, Zone):
+            return False
+        return self.name == other.name
+
+    def __hash__(self):
+        return f"{self.name}".__hash__()
+
+    def __repr__(self):
+        return f"Zone<{self.name}>"
+
+
+class Record(object):
+    """
+    Record represents a DNS record
+
+    :param zone: the zone the record belongs to
+    :param name: the record name
+    :param data: the record data
+    :param awsdata: the original aws data for a dns record
+    :type zone: Zone
+    :type name: str
+    :type data: dict
+    :type awsdata: dict
+    """
+
+    SUPPORTED_TYPES = ['A', 'CNAME', 'NS', 'TXT']
+
+    def __init__(self, zone, name, data={}, awsdata={}):
+        self._zone = zone
+        self._name = _octal_replace(name).lower().rstrip('.')
+
+        try:
+            self._type = data['type']
+            self._ttl = int(data['ttl'])
+        except KeyError as e:
+            raise InvalidRecordData(f"missing key {e} in Record data")
+
+        if self._type not in self.SUPPORTED_TYPES:
+            raise InvalidRecordType(f"Type {self._type} is not supported")
+
+        self._values = data.get('values', [])
+        self._data = data
+        self._awsdata = awsdata
+
+    @property
+    def data(self):
+        """Returns the record data"""
+        return self._data
+
+    @property
+    def fqdn(self):
+        """Returns the record's FQDN"""
+        return f"{self.name}.{self.zone.name}"
+
+    @property
+    def name(self):
+        """Returns the record name"""
+        return self._name
+
+    @property
+    def awsdata(self):
+        """Returns the record's awsdata"""
+        return self._awsdata
+
+    @property
+    def ttl(self):
+        """Returns the record TTL"""
+        return self._ttl
+
+    @property
+    def type(self):
+        """Returns the record type"""
+        return self._type
+
+    @property
+    def zone(self):
+        """Returns the parent zone"""
+        return self._zone
+
+    @property
+    def values(self):
+        """Returns the values for the record"""
+        return sorted(self._values)
+
+    def add_targets(self, values):
+        """
+        Add a list of targets to the record
+
+        :param values: a list of values to add
+        :type values: list of str
+        """
+        if not isinstance(values, list):
+            raise TypeError(f"expected list, received {type(values)}")
```

Comment:
You should net really care about types. You already documented the expected type, so now just use it.
If you want to be really nice, you would:
```
self._values.extend(list(values))
```
And let it beak would it not be convertible to list.

Reply:
fixed in 2febd00d2d18d255620036ff270c10bedb0fefc1